app/src/main/java/com/kks/nimblesurveyjetpackcompose/repo/home/HomeRepoImpl.kt

```diff
@@ -40,9 +41,7 @@ class HomeRepoImpl @Inject constructor(
             is ResourceState.Error -> emit(ResourceState.Error(apiResult.error))
             else -> emit(ResourceState.NetworkError)
         }
-    }.catch { error ->
-        emit(ResourceState.Error(error.message.orEmpty()))
-    }
+    }.catchError()
```

Comment:
Why not replace the `flow {}` method with a method to catch errors entirely? [Something like this](https://github.com/nimblehq/android-templates/blob/5b611f3ca5d78e6363bd8693c955a346a589b6f6/template/data/src/main/java/co/nimblehq/template/data/extensions/ResponseMapping.kt#L20) 😉 What do you think? 🤔

Reply:
@ryan-conway I am catching in the `RepoImpl` using `safeApiCall` extension and when I try using [something like this](https://github.com/nimblehq/android-templates/blob/5b611f3ca5d78e6363bd8693c955a346a589b6f6/template/data/src/main/java/co/nimblehq/template/data/extensions/ResponseMapping.kt#L20), it doesn't make a huge difference for me. So I am planning to leave it like that.

Reply:
Understood, just know that we will need to add that extension method to _all_ flow calls 🤔

Reply:
@ryan-conway Yes, I did.